fix(sharding): sharding guide overhaul (#629)
discordjs/guide: guide/sharding/extended.md

<branch version="11.x">

```js
function findEmoji(nameOrID) {

> 	const emoji = this.emojis.get(query) || this.emojis.find(e => e.name.toLowerCase() === nameOrID.toLowerCase());
> 	if (!emoji) return null;
> 	return emoji;

Reviewer: ```suggestion
	return this.emojis.get(nameOrID) || this.emojis.find(e => e.name.toLowerCase() === nameOrID.toLowerCase());
```

<branch version="12.x">

```js
function findEmoji(nameOrID) {

> 	const emoji = this.emojis.cache.get(nameOrID) || this.emojis.cache.find(e => e.name.toLowerCase() === nameOrID.toLowerCase());
> 	if (!emoji) return null;
> 	return emoji;

Reviewer: ```suggestion
	return this.emojis.cache.get(nameOrID) || this.emojis.cache.find(e => e.name.toLowerCase() === nameOrID.toLowerCase());
```

]
```

While this result isn't *necessarily* bad or incorrect, it's simply a raw object that got `JSON.parse()`'d and `JSON.stringify()`'d over, so all of the circular references are gone. More importantly, The object is no longer a true <branch version="11.x" inline>`Emoji`</branch><branch version="12.x" inline>`GuildEmoji`</branch> object as provided by discord.js. *This means none of the convenience methods usually provided to you are available.* If this is a problem for you, you will want to handle the item *inside* the `broadcastEval`. Conveniently, the `findEmoji` function will be ran inside it, so you should execute your relevant methods there, before the object leaves the context.

<branch version="11.x">

```js
function findEmoji(nameOrID) {
	const emoji = this.emojis.get(nameOrID) || this.emojis.find(e => e.name.toLowerCase() === nameOrID.toLowerCase());

> 	if (!emoji) return null;
> 	// If you wanted to delete the emoji with Discord.js, this is where you would do it. Otherwise, don't include this code.

Reviewer: ```suggestion
	if (!emoji) return;
	// If you wanted to delete the emoji with discord.js, this is where you would do it. Otherwise, don't include this code.
```

	emoji.delete();
	return emoji;
}
```

</branch>

<branch version="12.x">

```js
function findEmoji(nameOrID) {
	const emoji = this.emojis.cache.get(nameOrID) || this.emojis.cache.find(e => e.name.toLowerCase() === nameOrID.toLowerCase());

> 	if (!emoji) return null;
> 	// If you wanted to delete the emoji with Discord.js, this is where you would do it. Otherwise, don't include this code.

Reviewer: ```suggestion
	if (!emoji) return;
	// If you wanted to delete the emoji with discord.js, this is where you would do it. Otherwise, don't include this code.
```

	emoji.delete();
	return emoji;
}
```

</branch>

> WIth all that said and done, usually we want to display the result, so here is how we can go about doing that:

Reviewer: ```suggestion
With all that said and done, you'll usually want to display the result, so here is how you can go about doing that:
```

+           // Locate a non falsy result, which will be the emoji in question
+           const foundEmoji = emojiArray.find(emoji => emoji);
+           if (!foundEmoji) return message.reply('I could not find such an emoji.');

> +           return message.reply(`I have found the emoji ${foundEmoji.id ? `<${foundEmoji.animated ? 'a' : ''}:${foundEmoji.name}:${foundEmoji.id}>` : foundEmoji.name}!`);

Reviewer: ```suggestion
+           return message.reply(`I have found the ${foundEmoji.id ? `<${foundEmoji.animated ? 'a' : ''}:${foundEmoji.name}:${foundEmoji.id}>` : foundEmoji.name} emoji!`);
```

+           // Locate a non falsy result, which will be the emoji in question
+           const foundEmoji = emojiArray.find(emoji => emoji);
+           if (!foundEmoji) return message.reply('I could not find such an emoji.');

> +           return message.reply(`I have found the emoji ${foundEmoji.animated ? `<${foundEmoji.identifier}>` : `<:${foundEmoji.identifier}>`}!`);

Reviewer: ```suggestion
+           return message.reply(`I have found the ${foundEmoji.animated ? `<${foundEmoji.identifier}>` : `<:${foundEmoji.identifier}>`} emoji!`);
```
